lib/pure/sugar.nim

```diff
@@ -364,3 +364,41 @@ when (NimMajor, NimMinor) >= (1, 1):
         of "bird": "word"
         else: d
     assert z == @["word", "word"]
+
+proc splitDefinition*(def: NimNode): tuple[lhs: NimNode, rhs: NimNode, exported: bool] {.since: (1,1).} =
+  ## allows library constructs such as:
+  ## `byAddr: a2=expr`
+  ## `byAddr: a2*=expr` (to indicate `export`)
```

Comment:
Sorry, only now I understand you use ``*=`` and do not mean "multiply assign". That's unacceptable design. And why support exporting these aliases in the first place? The construct is safer than ``addr`` but not safe enough to expose these in your APIs, consider:

```nim

byAddr: x = myseq[1]
setLen(myseq, 0)
...
```

And why not support also ``byAddr x = expr`` without the colon?

Reply:
> And why not support also byAddr x = expr without the colon?

this can't work due to parsing rules; in `byAddr x = expr`, only `x` is passed to byAddr, not `x = expr`

; will reply to other point later

Reply:
fixed:
* `*=` now doesn't mean export but instead just forwards the operator to the caller that can handle it as they want
* if user wants to export, he can now do this: `byAddr: x0 = x[0]; export x0` , with all the caveats regarding memory safety, but at least that's now just orthogonal language features interacting normally